From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 19 Feb 2026 04:35:45 +0100
Subject: document-portal: Actually make use of an error on spawn

We allocate an error for spawning, but we do not use it at all.

Let's add some debug messages at least

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1764
---
 document-portal/document-portal.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/document-portal/document-portal.c b/document-portal/document-portal.c
index 1fa59e1..b702b1e 100644
--- a/document-portal/document-portal.c
+++ b/document-portal/document-portal.c
@@ -657,6 +657,7 @@ app_has_file_access (const char *target_permissions_id,
 
       if (!res)
         {
+          g_debug ("snap file-access check failed: %s", error->message);
           /* Fallback access method has no support for snaps. */
           return FALSE;
         }
@@ -666,6 +667,9 @@ app_has_file_access (const char *target_permissions_id,
       /* First we try flatpak info --file-access=PATH APPID, which is supported on new versions */
       arg = g_strdup_printf ("--file-access=%s", path);
       res = xdp_spawn (&error, "flatpak", "info", arg, target_permissions_id, NULL);
+
+      if (!res)
+        g_debug ("flatpak file-access check failed: %s", error->message);
     }
 
   if (res)
